src/phoenix/experimental/evals/retrievals.py

```diff
@@ -0,0 +1,82 @@
+"""
+Helper functions for evaluating the retrieval step of retrieval-augmented generation.
+"""
+
+from typing import List, Optional, Tuple
+
+import openai
+from tenacity import (
+    retry,
+    stop_after_attempt,
+    wait_random_exponential,
+)
+
+_EVALUATION_SYSTEM_MESSAGE = (
+    "You will be given a query and a reference text. "
+    "You must determine whether the reference text contains an answer to the input query. "
+    'Your response must be single word, either "relevant" or "irrelevant", '
+    "and should not contain any text or characters aside from that word. "
+    '"irrelevant" means that the reference text does not contain an answer to the query. '
+    '"relevant" means the reference text contains an answer to the query.'
+)
+_QUERY_CONTEXT_PROMPT_TEMPLATE = """# Query: {query}
+
+# Reference: {reference}
+
+# Answer ("relevant" or "irrelevant"): """
+
+
+def compute_precisions_at_k_and_relevance_classifications(
+    query_texts: List[str],
+    list_of_retrieved_document_texts: List[List[str]],
+    model_name: str = "gpt-4",
+) -> Tuple[List[List[Optional[float]]], List[List[Optional[bool]]]]:
```

Comment:
Lots of optionals and lists of lists of lists. It's hard to groc what the return signature is here. Can you make hte return signature grocable, either via type aliases or by returning a structure that is parsable?

Reply:
Done.